pallets/parachain-staking/src/lib.rs

```diff
@@ -0,0 +1,1870 @@
+            });
+            <InflationConfig<T>>::put(config);
+            Ok(().into())
+        }
+        #[pallet::weight(<T as Config>::WeightInfo::set_parachain_bond_account())]
+        /// Set the account that will hold funds set aside for parachain bond
+        pub fn set_parachain_bond_account(
+            origin: OriginFor<T>,
+            new: T::AccountId,
+        ) -> DispatchResultWithPostInfo {
+            T::MonetaryGovernanceOrigin::ensure_origin(origin)?;
+            let ParachainBondConfig {
+                account: old,
+                percent,
+            } = <ParachainBondInfo<T>>::get();
+            ensure!(old != new, Error::<T>::NoWritingSameValue);
+            <ParachainBondInfo<T>>::put(ParachainBondConfig {
+                account: new.clone(),
+                percent,
+            });
+            Self::deposit_event(Event::ParachainBondAccountSet { old, new });
+            Ok(().into())
+        }
+        #[pallet::weight(<T as Config>::WeightInfo::set_parachain_bond_reserve_percent())]
+        /// Set the percent of inflation set aside for parachain bond
+        pub fn set_parachain_bond_reserve_percent(
+            origin: OriginFor<T>,
+            new: Percent,
+        ) -> DispatchResultWithPostInfo {
+            T::MonetaryGovernanceOrigin::ensure_origin(origin)?;
+            let ParachainBondConfig {
+                account,
+                percent: old,
+            } = <ParachainBondInfo<T>>::get();
+            ensure!(old != new, Error::<T>::NoWritingSameValue);
+            <ParachainBondInfo<T>>::put(ParachainBondConfig {
+                account,
+                percent: new,
+            });
+            Self::deposit_event(Event::ParachainBondReservePercentSet { old, new });
+            Ok(().into())
+        }
+        #[pallet::weight(<T as Config>::WeightInfo::set_total_selected())]
+        /// Set the total number of collator candidates selected per round
+        /// - changes are not applied until the start of the next round
+        pub fn set_total_selected(origin: OriginFor<T>, new: u32) -> DispatchResultWithPostInfo {
+            frame_system::ensure_root(origin)?;
+            ensure!(
+                new >= T::MinSelectedCandidates::get(),
+                Error::<T>::CannotSetBelowMin
+            );
+            let old = <TotalSelected<T>>::get();
+            ensure!(old != new, Error::<T>::NoWritingSameValue);
+            ensure!(
+                new <= <Round<T>>::get().length,
+                Error::<T>::RoundLengthMustBeAtLeastTotalSelectedCollators,
+            );
+            <TotalSelected<T>>::put(new);
+            Self::deposit_event(Event::TotalSelectedSet { old, new });
+            Ok(().into())
+        }
+        #[pallet::weight(<T as Config>::WeightInfo::set_collator_commission())]
+        /// Set the commission for all collators
+        pub fn set_collator_commission(
+            origin: OriginFor<T>,
+            new: Perbill,
```

Comment:
#[pallet::compact]
and please check 0% or 101%, not sure 101% is possible.

Reply:
Percent/Perbill are bound-safe types, they _cannot_ be outside of 0-100

Reply:
so the frontend would fail trying to create this type when submitting the extrinsic ( or clamp the value to the correct range )